### ISSUE TITLE
componentWillMount not triggering with pagination

### ISSUE THREAD

**Issue by phamj88** [AllenFang/react-bootstrap-table#1110]
Hi!

Thanks again for the great tool.
I am using version 3.0.0
I am running into an issue where componentWillMount is not triggering with pagination.
I will try to explain this as best as possible.

I have a table that uses expandableRows.
The table contains lets say contains a hundred elements with pagination turned on with 10 per page.
The expandComponent returns a component that has a componentWillMount():
`
componentWillMount(){
    const id = this.props.row.id;
    console.log(taskid)
}
`

When I load the page the first 10 elements of page will do console.log(taskid).
Then the rest never do. If I click through the pagination, componentWillMount is not triggered.
But if i change the sizePerPage to all or 100 it will console log all the ones that are displayed.

This is an issue for me because I am doing async calls in these expandComponents. And I use componentWillMount to initialize my state, else it errors.

TLDR: componentWillMount only triggers for sizePerPageList initial value and not the other pages.

Thanks, Please let me know if you need any more info.

**Comment by AllenFang** [comment]
@phamj88, thanks for your clearly explanation, I'll check this out, 👍

**Comment by AllenFang** [comment]
Fixed on `v3.1.0`, check it out. 👍

**Comment by phamj88** [comment]
@AllenFang 
Thanks Allen for the quick fix!
You're the best!
